core/flow/rule.go

```diff
@@ -102,6 +105,16 @@ type Rule struct {
 	// If the StatIntervalInMs user specifies can not reuse the global statistic of resource,
 	// 		sentinel will generate independent statistic structure for this rule.
 	StatIntervalInMs uint32 `json:"statIntervalInMs"`
+
+	// adaptive flow control algorithm related parameters
+	// limitation: LowMemUsageThreshold > HighMemUsageThreshold && MemHighWaterMark > MemLowWaterMark
+	// if the current memory usage is less than or equals to MemLowWaterMark, threshold == LowMemUsageThreshold
+	// if the current memory usage is more than or equals to MemHighWaterMark, threshold == HighMemUsageThreshold
+	// if  the current memory usage is in (MemLowWaterMark, MemHighWaterMark), threshold is in (HighMemUsageThreshold, LowMemUsageThreshold)
+	LowMemUsageThreshold  int64 `json:"lowMemUsageThreshold"`
+	HighMemUsageThreshold int64 `json:"highMemUsageThreshold"`
+	MemLowWaterMark       int64 `json:"memLowWatermark"`
+	MemHighWaterMark      int64 `json:"memHighWatermark"`
```

Comment:
Add a unit suffix may be better, like `KB`, `MB`.

Reply:
Will use Bytes and provide convenient MB KB constant